backend/src/api/transaction-utils.ts

```diff
@@ -50,7 +59,31 @@ class TransactionUtils {
       feePerVsize: feePerVbytes,
       effectiveFeePerVsize: feePerVbytes,
     }, transaction);
-    if (!transaction.status.confirmed) {
+    if (!transaction?.status?.confirmed) {
+      transactionExtended.firstSeen = Math.round((new Date().getTime() / 1000));
+    }
+    return transactionExtended;
+  }
+
+  public extendMempoolTransaction(transaction: IEsploraApi.Transaction): MempoolTransactionExtended {
+    const vsize = Math.ceil(transaction.weight / 4);
+    const fractionalVsize = (transaction.weight / 4);
+    const sigops = this.countSigops(transaction);
+    // https://github.com/bitcoin/bitcoin/blob/e9262ea32a6e1d364fb7974844fadc36f931f8c6/src/policy/policy.cpp#L295-L298
+    const adjustedVsize = Math.max(fractionalVsize, sigops *  5); // adjusted vsize = Max(weight, sigops * bytes_per_sigop) / witness_scale_factor
```

Comment:
Since sigops * 5 can only be an integer, but fractionalVsize can be a non-integer, if we imagine the case where fractionalVsize is between the sigops * 5 and (sigops * 5) - 1, the sigops * 5 will be chosen, but if you ceil fractionalVsize, it will be equal and won't matter.

So it would be better to use vsize which is an integer, so that adjustedVsize is also an integer.

Reply:
Or maybe the ceil should be done inside the Object.assign below.